clientv1.go

```diff
@@ -26,14 +26,71 @@ type ClientV1 struct {
 	defaultInterceptors []connect.Interceptor
 }
 
+type ClientV1ConnConfig struct {
+	// ExternalURL is the configured default external ExternalURL of the relevant Sourcegraph
+	// Accounts instance.
+	ExternalURL string
+	// APIURL is the URL to use for Sourcegraph Accounts API interactions. This
+	// can be set to some internal URLs for private networking. If this is nil,
+	// the client will fall back to URL instead.
+	APIURL *string
+	// Client credentials representing this Sourcegraph Accounts client
+	ClientID     string
+	ClientSecret string
+}
+
+// envGetter is a helper interface for getting environment variables based on
+// the MSP runtime environment, matching the type
+// github.com/sourcegraph/sourcegraph/lib/managedservicesplatform/runtime/contract.Env.
+type envGetter interface {
+	// Get returns the value with the given name. If no value was supplied in the
+	// environment, the given default is used in its place. If no value is available,
+	// an error is added to the validation errors list.
+	Get(name, defaultValue, description string) string
+	// GetOptional returns the value with the given name, or nil if no value is
+	// available.
+	GetOptional(name, description string) *string
+}
+
+const defaultExternalURL = "https://accounts.sourcegraph.com"
+
+// NewClientV1ConnectionConfigFromEnv initializes configuration for connecting
+// to Sourcegraph Accounts using default standards for loading environment
+// variables. This allows the Core Services team to more easily configure access.
+func NewClientV1ConnectionConfigFromEnv(env envGetter) (c ClientV1ConnConfig) {
+	c.ExternalURL = env.Get("SAMS_URL", defaultExternalURL, "External URL of the connected SAMS instance")
```

Comment:
@unknwon I don't think we will use this internally within the SDK, but since the config type is intended to be used, I think it makes sense to make this available to consumers which might want it for e.g. redirection.